src/orquestra/sdk/_base/_qe/_qe_runtime.py

```diff
@@ -353,10 +352,12 @@ def _find_packed_artifact_id(
 ) -> ArtifactNodeId:
     output_ids = wf_def.task_invocations[inv_id].output_ids
     artifact_nodes = (wf_def.artifact_nodes[id] for id in output_ids)
-    packed: ArtifactNode = _find_first(
-        lambda node: node.artifact_index is None, artifact_nodes
-    )
-    return packed.id
+    packed_nodes = [n for n in artifact_nodes if n.artifact_index is None]
+    assert len(
+        packed_nodes
+    ), f"Task invocation should have exactly 1 packed output. {inv_id} has {len(packed_nodes)}: {packed_nodes}"  # noqa: E501
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
**issue:** ‏all non-zero values are truthy
```suggestion
    assert len(
        packed_nodes
    ) == 1, f"Task invocation should have exactly 1 packed output. {inv_id} has {len(packed_nodes)}: {packed_nodes}"  # noqa: E501
```